File: solutions/swift/perfect-numbers/2/Sources/PerfectNumbers/PerfectNumbers.swift
enum Classification { case perfect, abundant, deficient }
enum ClassificationError: Error { case invalidInput }

func classify(number: Int) throws -> Classification {
  guard number > 0 else { throw ClassificationError.invalidInput }
  var a = 0
  for factor in 1..<(number / 2) + 1 {
    if number % factor == 0 { a += factor }
    if a > number { return .abundant }
  }
  return a < number ? .deficient : .perfect
}

File: solutions/zig/perfect-numbers/4/perfect_numbers.zig
pub const Classification = enum {
    deficient,
    perfect,
    abundant,
};

/// Asserts that `n` is nonzero.
pub fn classify(n: u64) Classification {
    var a: usize = 0;
    for (1..(n / 2) + 1) |f| {
        if (n % f == 0) a += f;
    }

    if (a < n) return .deficient;
    if (a > n) return .abundant;
    return .perfect;
}
